frontend/src/modules/location/views/locationTitleView.js

```diff
@@ -1,41 +1,63 @@
 // LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
 define(function(require) {
-
-	var Origin = require('core/origin');
-	var Backbone = require('backbone');
-
-	var LocationTitleView = Backbone.View.extend({
-
-		el: '.location-title',
-
-		initialize: function() {
-			this.listenTo(Origin, 'location:title:update', this.render);
-			this.listenTo(Origin, 'location:title:hide', this.onHideTitle);
-			this.render();
-		},
-
-		render: function(data) {
-		    var template = Handlebars.templates[this.constructor.template];
-		    this.$el.html(template(data));
-		    _.defer(_.bind(function() {
-		    	this.postRender();
-		    }, this));
-		    return this;
-		},
-
-		postRender: function() {
-			this.$el.removeClass('display-none');
-			Origin.trigger('location:title:postRender', this);
-		},
-
-		onHideTitle: function() {
-			this.$el.addClass('display-none');
-		}
-
-	}, {
-		template: 'locationTitle'
-	});
-
-	return LocationTitleView;
-
-});
\ No newline at end of file
+  var _ = require('underscore');
+  var Backbone = require('backbone');
+  var Origin = require('core/origin');
+
+  var EditorDataLoader = require('modules/editor/global/editorDataLoader');
+
+  var LocationTitleView = Backbone.View.extend({
+    el: '.location-title',
+
+    initialize: function() {
+      this.listenTo(Origin, {
+        'location:title:update': this.render,
+        'location:title:hide': this.onHideTitle
+      });
+      EditorDataLoader.waitForLoad(_.bind(this.render, this));
+    },
+
+    render: function(data) {
+      var template = Handlebars.templates[this.constructor.template];
```

Comment:
Can we include the elements title if available?

![image](https://user-images.githubusercontent.com/11990374/51987114-034a1780-24a2-11e9-8da7-83e61556b8c5.png)

If it is too long, we can also set it as a `title` attribute to show it on hover.